### ISSUE TITLE
Library does not support hermes on android?

### ISSUE THREAD

**Issue by pani7** [wix/react-native-notifications#455]
Hi all, does this library support hermes on android? I am using version 3.1.1 and i keep getting: Property proxy does not exist, js engine: hermes

**Comment by simonasdev** [comment]
It appears not. Since hermes specifically excludes Proxy objects from support: https://github.com/facebook/hermes/blob/master/doc/Features.md#excluded-from-support

**Comment by AndrewMorsillo** [comment]
Hermes supports Proxy now, yay! Unfortunately it's not quite ready for use yet https://github.com/facebook/hermes/issues/33#issuecomment-576388658

**Comment by radko93** [comment]
I wonder if it is really important to use a Proxy in this library? It does not seem like a crucial functionality.

**Comment by robertmaples** [comment]
If proxy is non-essential, excluding it would be great. As mentioned above there has been a build released for hermes that includes proxy, but is not compatible with an available version of RN (no timeline on when that'll be available either).

**Comment by orhanoksuzz** [comment]
> I wonder if it is really important to use a Proxy in this library? It does not seem like a crucial functionality.
> I just commented out the proxy code in `NotificationsIOS` and `NotificationsAndroid` and it works fine. I might make a PR that refactors the code if that's what we want to do.

Could you pr edit the code? It would be great to use this library with Hermes

**Comment by poocart** [comment]
Fixed by adding `proxy-polyfill` to main project and `import 'proxy-polyfill'` on top of `react-native-notifications` lib files `src/NotificationsAndroid.ts` and `src/NotificationsIOS.ts`, then rebuilding sources and patching lib with `patch-package` lib. If by any change no one will come up with PR I'll fork this and make PR with my changes.

**Comment by from74** [comment]
> Fixed by adding `proxy-polyfill` to React Native project `package.json` and `import 'proxy-polyfill'` on top of `react-native-notifications` lib files `src/NotificationsAndroid.ts` and `src/NotificationsIOS.ts`, then rebuilding lib's sources and patching lib with `patch-package`.
> 
> If by any change no one will come up with PR I'll fork this and make PR with my changes.

I'm very sorry, but not understand one thing.. How I can rebuild lib's sources? What should I do for this?

**Comment by paintingwithcode** [comment]
@from74, after you make the changes to `src/NotificationsIOS.ts` and `src/NotificationsAndroid.ts`, cd into the package folder in your `node_modules` and run `yarn && yarn build`. It will recompile the `dist` folder with the changes you made in `src`. 

Only after rebuilding the `dist` folder, should you run `npx patch-package react-native-notifications` because otherwise the app will continue to compile with the pre-patched code.

**Comment by axebelk** [comment]
> @from74, after you make the changes to `src/NotificationsIOS.ts` and `src/NotificationsAndroid.ts`, cd into the package folder in your `node_modules` and run `yarn && yarn build`. It will recompile the `dist` folder with the changes you made in `src`.
> 
> Only after rebuilding the `dist` folder, should you run `npx patch-package react-native-notifications` because otherwise the app will continue to compile with the pre-patched code.

After done all these changes, I have got " Error: Unable to resolve module `warnOnce` from " error

**Comment by owinter86** [comment]
I can confirm that `hermes-engine@v0.4.2-rc1` which has proxy support works with this package.

https://twitter.com/HermesEngine/status/1245136667414913024

**Comment by poocart** [comment]
> Fixed by adding `proxy-polyfill` to React Native project `package.json` and `import 'proxy-polyfill'` on top of `react-native-notifications` lib files `src/NotificationsAndroid.ts` and `src/NotificationsIOS.ts`, then rebuilding lib's sources and patching lib with `patch-package`.
> 
> If by any change no one will come up with PR I'll fork this and make PR with my changes.

For those who still have issues – https://github.com/wix/react-native-notifications/pull/553. This approach works in my target project.

**Comment by gavrichards** [comment]
> > @from74, after you make the changes to `src/NotificationsIOS.ts` and `src/NotificationsAndroid.ts`, cd into the package folder in your `node_modules` and run `yarn && yarn build`. It will recompile the `dist` folder with the changes you made in `src`.
> > Only after rebuilding the `dist` folder, should you run `npx patch-package react-native-notifications` because otherwise the app will continue to compile with the pre-patched code.
> 
> After done all these changes, I have got " Error: Unable to resolve module `warnOnce` from " error

I attempted the change explained above too but then also got the same 'warnOnce' error message. I decided to roll it back so I'm back to the Hermes proxy issue again. Is there a better solution on the horizon?

**Comment by owinter86** [comment]
@gavrichards refer to my previous comment, you can install the Hermes 4.2 release candidate which has proxy support. I have this running in production currently.

**Comment by tomoima525** [comment]
@owinter86 do you mean we explicitly install `hermes-engine` to the project package?

@gavrichards @axebelk this issue happens because react-native-notifications looks at `react-native@0.60.5` which has the different package directory of `warnOnce` (it's under `react-native-implementation/util`)

I was not able to fix this issue by updating react-native inside react-native-notifications to 0.61.5. It caused another issue. Currently the only way to fix this issue(other than updating hermes-engine which I haven't tested yet) is to remove Proxy. Basically you need to add `Platform.OS === 'android'` to NotificationAndroid.js and `Platform.OS === 'ios'` to NotificationIOS.js

```
Object.defineProperty(exports, "__esModule", { value: true });
const react_native_1 = require("react-native");

class NotificationsIOS {
    constructor(commands, eventsRegistry) {
        this.commands = commands;
        this.eventsRegistry = eventsRegistry;
    }

    /**
    * Request permissions to send remote notifications
    */
    registerRemoteNotifications() {
        if (react_native_1.Platform.OS === 'ios') {
            return this.commands.requestPermissions();
        }
        return {};
    }

    /**
    * Unregister for all remote notifications received via Apple Push Notification service
    */
    abandonPermissions() {
        if (react_native_1.Platform.OS === 'ios') {
            return this.commands.abandonPermissions();
        }
        return {};
    }

    /**
    * registerPushKit
    */
    registerPushKit() {
        if (react_native_1.Platform.OS === 'ios') {
            return this.commands.registerPushKit();
        }
        return {};
    }

    /**
    * getBadgeCount
    */
    getBadgeCount() {
        if (react_native_1.Platform.OS === 'ios') {
            return this.commands.getBadgeCount();
        }
        return {};
    }

    /**
    * setBadgeCount
    * @param count number of the new badge count
    */
    setBadgeCount(count) {
        if (react_native_1.Platform.OS === 'ios') {
            return this.commands.setBadgeCount(count);
        }
        return {};
    }

    /**
    * cancelAllLocalNotifications
    */
    cancelAllLocalNotifications() {
        if (react_native_1.Platform.OS === 'ios') {
            this.commands.cancelAllLocalNotifications();
        }
    }

    /**
    * checkPermissions
    */
    checkPermissions() {
        if (react_native_1.Platform.OS === 'ios') {
            return this.commands.checkPermissions();
        }
        return {};
    }

    /**
    * removeDeliveredNotifications
    * @param identifiers Array of notification identifiers
    */
    removeDeliveredNotifications(identifiers) {
        if (react_native_1.Platform.OS === 'ios') {
            return this.commands.removeDeliveredNotifications(identifiers);
        }
        return {};
    }

    /**
    * getDeliveredNotifications
    */
    getDeliveredNotifications() {
        if (react_native_1.Platform.OS === 'ios') {
            return this.commands.getDeliveredNotifications();
        }
        return {};
    }

    /**
    * Obtain the events registry instance
    */
    events() {
        if (react_native_1.Platform.OS === 'ios') {
            return this.eventsRegistry;
        }
        return {};
    }
}
exports.NotificationsIOS = NotificationsIOS; 
```

```
Object.defineProperty(exports, "__esModule", { value: true });
const react_native_1 = require("react-native");

class NotificationsAndroid {
    constructor(commands) {
        this.commands = commands;
    }

    /**
    * Refresh FCM token
    */
    registerRemoteNotifications() {
        if (react_native_1.Platform.OS === 'android') {
        this.commands.refreshToken();
        }
    }
}
exports.NotificationsAndroid = NotificationsAndroid;
```

**Comment by owinter86** [comment]
Yep just yarn/npm install it and that’s it, delete your build folder and rebuild.

**Comment by radko93** [comment]
@owinter86 this might now work, hermes 0.4 might require RN 0.62

**Comment by gavrichards** [comment]
I can confirm `npm i hermes-engine@0.4.2-rc1` did the trick for me.

The release notes for [Hermes 0.5.0](https://github.com/facebook/hermes/releases/tag/v0.5.0) mention it's intended for React Native 0.63.0, so looks like we'll need to wait for that for a proper fix that doesn't require installing Hermes independently in your project, or for this library to be updated with a fix.

**Comment by SmartArray** [comment]
> I can confirm `npm i hermes-engine@0.4.2-rc1` did the trick for me.
> 
> The release notes for [Hermes 0.5.0](https://github.com/facebook/hermes/releases/tag/v0.5.0) mention it's intended for React Native 0.63.0, so looks like we'll need to wait for that for a proper fix that doesn't require installing Hermes independently in your project, or for this library to be updated with a fix.

So let me try to sum it up:

```
rm -rf node_modules
npm install hermes-engine@0.4.2-rc1 
npm install
```

**Note:** In my case, upgrading to React-Native 0.62 was required in order to make the Proxy work.

**Comment by comvenger-brandon** [comment]
> @gavrichards refer to my previous comment, you can install the Hermes 0.4.2 release candidate which has proxy support. I have this running in production currently.

We noticed random crashes in production with the rc1.

**Comment by sarthakpranesh** [comment]
I can confirm this issue remains in React Native v0.63.0 using react-native-notifications v3.3.4 and outputs `ReferenceError: Property 'Proxy' doesn't exist, js engine: hermes`

**Comment by owinter86** [comment]
@sarthakpranesh you can use the v5 Hermes engine version with RN version 0.63.X

https://github.com/facebook/hermes/releases/tag/v0.5.2-rc1 

or wait for RN 0.64 which will have the new Hermes engine with proxy enabled by default.

**Comment by sarthakpranesh** [comment]
@owinter86 I see, thanks for the help :+1:

**Comment by ebaynaud** [comment]
For RN 0.63 edit package.json like this:
```
...
  "dependencies": {
...
    "hermes-engine": "0.5.2-rc1"
...
  },
  "resolutions": {
...
    "react-native/hermes-engine": "0.5.2-rc1"
...
  },
...
```

**Comment by Jpunt** [comment]
I'd like to say that using release candidates for something substantial as a JS-engine is not to be advised. 

To the people at Wix: Would it be a huge ordeal to refactor out Proxy?

**Comment by owinter86** [comment]
@jpunt Hermes supports proxy now and will be included in react native release 0.64, you can use the beta versions if you are on previous react native versions as per the comments in this thread.

**Comment by Jpunt** [comment]
Yeah, my point exactly: When building production level apps, it's unwise to use beta versions of these kinds of foundation libs.

**Comment by pastean** [comment]
> @sarthakpranesh you can use the v5 Hermes engine version with RN version 0.63.X
> 
> https://github.com/facebook/hermes/releases/tag/v0.5.2-rc1
> 
> or wait for RN 0.64 which will have the new Hermes engine with proxy enabled by default.

Confirming that upgrading to 0.64 fixed the issue. For me it had nothing to do with this library, had the issue from somewhere else.
